mkosi

```diff
@@ -564,6 +564,18 @@ def mount_image(args, workspace, loopdev, root_dev, home_dev, srv_dev, root_read
 
             umount(root)
 
+@complete_step("Assigning hostname")
+def assign_hostname(args, workspace):
+    root = os.path.join(workspace, "root")
+    hostname_path = os.path.join(root, "etc/hostname")
+    
+    if not args.hostname:
+        if os.path.isfile(hostname_path):
+            os.remove(hostname_path)
+    else:
+        with open(hostname_path, "w+") as f:
```

Comment:
given that there's a good chance that /etc/hostname is already a symlink to /run/systemd/resolve/resolv.conf I think it would make sense to explicitly remove the file before we write to it, otherwise the open() would just follow the symlink and that's not what we want here...